src/AbstractFrontendDataProvider.php

```diff
@@ -10,7 +10,7 @@ abstract class AbstractFrontendDataProvider
     /**
      * @var string Optional key that will be used when this provider is json encoded
      */
-    protected $key;
+    protected string $key;
```

Comment:
We might have to add a default (null or empty string) here, since PHP7.4 requires properties to be initialised before they can be accessed, see https://madewithlove.com/typed-property-must-not-be-accessed-before-initialization/